R/zchunk_LA154.Transport.R

```diff
@@ -32,71 +31,215 @@ module_gcam.usa_LA154.Transport_DISABLED <- function(command, ...) {
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+
+      # Silence package notes
+      GCAM_region_ID <- UCD_sector <- mode <- size.class <- UCD_technology <- UCD_fuel <- fuel <- EIA_fuel <-
+        year <- value <- EIA_sector <- . <- fuel_sector <- state <- sector <- value_state <- value_national <-
+        value_share <- value_mode <- NULL
+
+      # Calculate the state-wise percentages for each of EIA's sector/fuel combinations that is relevant for disaggregating
+      # nation-level transportation energy to the states
+
+      # This starting table is transportation energy consumption by GCAM region (and other variables)
+      # We will first subset this data for only the USA and values that are > 0 in the historical periods
+      L154.in_EJ_R_trn_m_sz_tech_F_Yh %>%
+        filter(year %in% HISTORICAL_YEARS, GCAM_region_ID == gcam.USA_CODE) %>% # Filter for the USA and for historical years only
+        filter(value != 0) %>% # Here any rows with value of 0 will be lost, even if other years of the same group are nonzero
+        # To get these rows back, the dataframe will be spread and gathered to reintroduce those rows with NAs, which will then be replaced with 0
+        spread(year, value) %>%
+        gather(year, value, -GCAM_region_ID, -UCD_sector, -mode, -size.class, -UCD_technology, -UCD_fuel, -fuel) %>%
+        mutate(year = as.integer(year)) %>%
+        replace_na(list(value = 0)) %>%
+        # Fuel and mode will be mapped to EIA fuel and sector
+        left_join_error_no_match(trnUCD_EIA_mapping, by = c("fuel", "mode")) ->
+        Transportation_energy_consumption
+
+      # From the full state database, state shares will be calculated based on relevant EIA sector and fuel combinations
+      # These shares will later be multipled by the transportation energy consumption data above
+      # We will create a list first, concatenating EIA-fuel and -sector, so as to selectively remove those pairs from the dataset
+      trnUCD_EIA_mapping %>%
+        mutate(fuel_sector = paste(EIA_fuel, EIA_sector)) %>%
+        .[["fuel_sector"]] %>%
+        unique() ->
+        list_fuel_sector
```

Comment:
`list_fuel_sector <- paste(trnUCD_EIA_mapping$EIA_fuel, trnUCD_EIA_mapping$EIA_sector) %>% unique` might be clearer and more compact? Up to you.